src/main/AuthenticationRequest.ts

```diff
@@ -98,6 +111,31 @@ export default class AuthenticationRequest {
       payload: verifiedJWT.payload as AuthenticationRequestPayload,
     };
   }
+
+  static async assertValidRegistration(verPayload: AuthenticationRequestPayload) {
+    if (verPayload.registration_uri && verPayload.registration) {
+      throw new Error(`${SIOPErrors.REG_OBJ_N_REG_URI_CANT_BE_SET_SIMULTANEOUSLY}`);
+    } else if (verPayload.registration_uri) {
+      try {
+        new URL(verPayload.registration_uri);
+      } catch (e) {
+        console.log(e);
+        throw new Error(`${SIOPErrors.REG_PASS_BY_REFERENCE_INCORRECTLY}`);
+      }
+    }
+
+    let regObj = verPayload.registration;
+
+    if (verPayload.registration_uri) {
+      regObj = (await getWithUrl(verPayload.registration_uri)) as unknown as RPRegistrationMetadataPayload;
+    }
+
+    if (regObj && !validate(regObj)) {
+      throw new Error('Registration data validation error: ' + JSON.stringify(validate.errors));
+    } else if (regObj?.subject_syntax_types_supported && regObj.subject_syntax_types_supported.length == 0) {
+      throw new Error(`${SIOPErrors.VERIFY_BAD_PARAMS}`);
+    }
+  }
```

Comment:
This is fetched only when the object is not passed by value. In order to do it somewhere else I will have to duplicate the all the checks to make sure the URL is there ... So I will be asserting/validating somewhere else...  which does not make sense at this ... It is requested to keep it here

Reply:
Yes I know. But it makes more sense to have the method doing the checks to only accept the object. Then have a seperate method either getting the object from the payload (by Value) or fetching it (by Reference). As next to doing the assertion of course we will need to use the object at a later point as well. Right now it might be fetching something then we discard the fetched data and a bit later we would have to fetch the exact same URL again.

So you have one method to extract the object or fetch it remotely. The end result is an object no matter whether it was passed by Value or by Object.
The next method is to actually verify the object content against the schema and perform some other checks on it

Reply:
fixed. in upcoming commit.